src/client/debugger/types.ts

```diff
@@ -16,39 +16,40 @@ export interface IKernelDebugAdapter {
 
 export const IDebuggingManager = Symbol('IDebuggingManager');
 export interface IDebuggingManager {
+    readonly onDoneDebugging: Event<void>;
     isDebugging(notebook: NotebookDocument): boolean;
     getDebugMode(notebook: NotebookDocument): KernelDebugMode | undefined;
     getDebugSession(notebook: NotebookDocument): Promise<DebugSession> | undefined;
     getDebugCell(notebook: NotebookDocument): NotebookCell | undefined;
 }
 
-export interface DebuggingDelegate {
+export interface IDebuggingDelegate {
     /**
      * Called for every event sent from the debug adapter to the client. Returns true to signal that sending the message is vetoed.
      */
-    willSendEvent(msg: DebugProtocolMessage): Promise<boolean>;
+    onWillSendEvent(msg: DebugProtocolMessage): Promise<boolean>;
```

Comment:
Usually I feel that events start with `on...`, this is a method and not an event. I don't know if your codebase has the same pattern, it's up to you

Reply:
true, that would be confusing, I'll put it back